### ISSUE TITLE
low level code example of hierarchical structure

### ISSUE THREAD

**Issue by chickenbestlover** [numenta/nupic#2933]
hi, im newbie to nupic. i'm working on training midi files(*.mid , a sort of music file) using low-level codes. 
Low-level codes mean just a raw sp and tp code, not network API or OPF.
I prefer to use low-level codes because it's easier for me to modify the codes.
I used a simple structure like this:
      raw midi data -> Encoder -> Spatial Pooler -> Temporal Pooler -> CLA Classifier -> prediction 
The result was quit awesome. the HTM successfully predicted the whole sequence with no error 

Then I wanted to change the structure to be hierarchical like this :
     raw midi data -> Encoder -> SP1 -> TP1  -> SP2 -> TP2 -> CLA Classifier -> prediction 

but I cannot implement the structure because i don't know how the layer 1 and layer 2 is linked. I already watched "hierarchy_network_demo.py", but the code just tells us "UniformLink". 
What does the term "UniformLink" mean? 

I think it's gonna be a strange architecture if TP1's output(array of cells) becomes the input of SP2, because in this hierarchy, layer 2 (SP2 and TP2) will have bigger column dimension than layer 1, which is somehow weird.

to sum up, my questions are :
   1. How they are linked between layers.
   2. Any hierarchy structure examples in low-level (not Network API, not OPF)

Any comments would be very helpful. Thank you.

**Comment by rhyolight** [comment]
Hi @chickenbestlover. I'm happy you are experimenting with NuPIC and getting good results! We certainly would like to help you answer your questions, but this Github Issue is not really the best place for it. Are you willing to sign up for our [general discussion NuPIC mailing list](http://lists.numenta.org/mailman/listinfo/nupic_lists.numenta.org) and ask your question there? We have over 1500 people on that list, many of whom would be interested in what you are doing.
